apps/nextjs/src/app/[locale]/manage/apps/_form.tsx

```diff
@@ -33,20 +39,31 @@ export const AppForm = (props: AppFormProps) => {
     },
   });
 
+  const handleSubmitAndCreateAnother = () => {
+    originalHandleSubmit(form.values, false, () => {
+      form.reset();
+    });
+  };
+
   return (
-    <form onSubmit={form.onSubmit(handleSubmit)}>
+    <form onSubmit={form.onSubmit((values) => originalHandleSubmit(values, true))}>
       <Stack>
         <TextInput {...form.getInputProps("name")} withAsterisk label={t("app.field.name.label")} />
-        <IconPicker initialValue={initialValues?.iconUrl} {...form.getInputProps("iconUrl")} />
+        <IconPicker {...form.getInputProps("iconUrl")} />
         <Textarea {...form.getInputProps("description")} label={t("app.field.description.label")} />
         <TextInput {...form.getInputProps("href")} label={t("app.field.url.label")} />
 
         <Group justify="end">
           <Button variant="default" component={Link} href="/manage/apps">
             {t("common.action.backToOverview")}
           </Button>
-          <Button type="submit" loading={isPending}>
-            {submitButtonTranslation(t)}
+          {buttonLabels.submitAndCreateAnother && (
+            <Button disabled={!form.isValid()} onClick={handleSubmitAndCreateAnother} loading={isPending}>
```

Comment:
IMO we should not disable a button and instead let the user click the button to validate the form (and show errors) I see the problem that we get as we have two submit buttons, can we fix this through an onclick that sets a varible